metadata-helper/src/signatures.rs

```diff
@@ -0,0 +1,101 @@
+use crate::AppState;
+
+use actix_files::NamedFile;
+use actix_web::HttpRequest;
+use commons::tracing::get_tracer;
+use commons::{self, format_request, Fallible, GraphError};
+use opentelemetry::trace::{mark_span_as_active, Tracer};
+use prometheus::{histogram_opts, Histogram, IntCounterVec, Opts, Registry};
+use std::collections::HashSet;
+use std::path::PathBuf;
+
+lazy_static! {
+    static ref SIGNATURES_INCOMING_REQS: IntCounterVec = IntCounterVec::new(
+        Opts::new("signatures_incoming_requests_total",
+        "Total number of incoming HTTP client request"),
+        &["uri_path"]
+    )
+    .unwrap();
+    // Histogram with custom bucket values for serving latency metric (in seconds), values are picked based on monthly data
+    static ref SIGNATURES_SERVE_HIST: Histogram = Histogram::with_opts(histogram_opts!(
+        "signatures_serve_duration_seconds",
+        "HTTP graph serving latency in seconds",
+        vec![0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1.0, 5.0]
+    ))
+    .unwrap();
+    // Supported Signatures Algo
+    static ref SUPPORTED_ALGO: HashSet<&'static str> = HashSet::from(["sha"]);
+}
+
+/// Register relevant metrics to a prometheus registry.
+pub(crate) fn register_metrics(registry: &Registry) -> Fallible<()> {
+    commons::register_metrics(registry)?;
+    registry.register(Box::new(SIGNATURES_INCOMING_REQS.clone()))?;
+    registry.register(Box::new(SIGNATURES_SERVE_HIST.clone()))?;
+    Ok(())
+}
+
+/// Serve Cincinnati graph requests.
```

Comment:
Does "graph requests" represent well what do we do here? "signature data requests" would be better perhaps?